tests/src/kakarot/instructions/test_system_operations.py

```diff
@@ -0,0 +1,229 @@
+import pytest
+from eth_account.account import Account
+from eth_utils import keccak
+from starkware.starknet.public.abi import get_selector_from_name
+
+from tests.utils.constants import CHAIN_ID
+from tests.utils.helpers import generate_random_private_key, pack_into_u64_words
+from tests.utils.syscall_handler import SyscallHandler
+from tests.utils.uint256 import int_to_uint256
+
+NONCE = 1
+MAGIC = 0x04
+CAIRO1_HELPERS_CLASS_HASH = 0xDEADBEEFABDE1
+
+
+@pytest.fixture
+def private_key():
+    return generate_random_private_key()
+
+
+@pytest.fixture
+def invoker_address(private_key):
+    return int(private_key.public_key.to_checksum_address(), 16)
+
+
+@pytest.fixture
+def message(invoker_address):
+    return {
+        "MAGIC": MAGIC.to_bytes(1, "big"),
+        "chainId": CHAIN_ID.to_bytes(32, "big"),
+        "nonce": NONCE.to_bytes(32, "big"),
+        "invokerAddress": invoker_address.to_bytes(32, "big"),
+        "commit": "commit".encode("utf-8").ljust(32, b"\x00"),
```

Comment:
you can do `b"commit"` directly. why do you pad here? the opcode is supposed to handle <32 commits right?

Reply:
Yeah, it's supposed to handle it.

Reply:
Bytes outside the range (in the event length is less than 97) are treated as if they had been zeroes.

So yes, the opcode is supposed to handle that. Then I need to dynamically read the message commit size to specify the input length.